src/patch-accessors.js

```diff
@@ -248,14 +251,13 @@ let InsideAccessors = {
     },
     set(text) {
       let content = this.localName === 'template' ? this.content : this;
-      clearNode(content);
-      let doc = domParser.parseFromString(text, 'text/html');
-      if (doc.head) {
-        insertDOMFrom(content, doc.head);
-      }
-      if (doc.body) {
-        insertDOMFrom(content, doc.body);
+      clearNode(this);
+      if (nativeInnerHTMLDesc && nativeInnerHTMLDesc.set) {
+        nativeInnerHTMLDesc.set.call(htmlContainer, text);
+      } else {
+        htmlContainer.innerHTML = text;
       }
+      insertDOMFrom(content, htmlContainer);
```

Comment:
`htmlContainer` could be a `<template>`, so that we could always set its `innerHTML` and just append the document fragment to the content
```js
const htmlContainer = document.createElement('template');
//...
clearNode(content);
htmlContainer.innerHTML = text;
content.appendChild(htmlContainer.content);